remote.go

```diff
@@ -0,0 +1,328 @@
+package imgutil
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/pkg/errors"
+)
+
+type RemoteImage struct {
+	keychain   authn.Keychain
+	RepoName   string
```

Comment:
I know these were already exported before we moved this over to this new repo, but is there any reason to expose RepoName, Image and PrevLayers? They seem like implementation details that should be encapsulated.

Reply:
Done.